### PR TITLE
Fix execution for throughput test

### PR DESCRIPTION
**Pull request by radek-starburst**
There is a bug that makes it impossible to save benchmark results in the benchto-service. This PR resolves it.

### CODE REVIEW

File: benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java [hunk 1]
```diff
@@ -249,7 +249,7 @@ private List<QueryExecutionResult> executeConcurrentQueries(Benchmark benchmark,
                         firstQuery = false;
```
Comment (raunaqmorarka) [inline benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java:249]: Please update your commit description to describe the problem scenario and the fix
E.g.
```
Benchmark results were not saved in benchto-service in XYZ scenario.
This is resolved by doing ABC.
```

File: benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java [hunk 2]
```diff
@@ -284,7 +284,7 @@ private QueryExecutionResult executeSingleQuery(
         QueryExecutionResult result;
         macroService.runBenchmarkMacros(benchmark.getBeforeExecutionMacros(), benchmark, connection);
 
-        if (!warmup) {
+        if (!warmup && !benchmark.isThroughputTest()) {
```
Comment (raunaqmorarka) [inline benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java:287]: Why wouldn't we want to report status for throughput test ? It deserves a code comment

Comment (radek-starburst) [reply benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java:287]: Because this is how throughput a test was implemented. The througput test has a 
[dedicated way](https://github.com/trinodb/benchto/blob/8051535c02e47a09d0b352d1d33f55492e9c7255/benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java#L212) to save results, especially to avoid a lot overhead of saving data per query. 

Please note that https://github.com/trinodb/benchto/blob/8051535c02e47a09d0b352d1d33f55492e9c7255/benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java#L252 `warmup` was hardoded to the `false` so the behaviour was not changed by my code. It was just fixed.

**Comment by przemekak** [comment]
Could you add E2E throughput test to benchto-it? It is kind of different, uses different reporting so I believe it deserves one test there.

**Comment by radek-starburst** [comment]
> Could you add E2E throughput test to benchto-it? It is kind of different, uses different reporting so I believe it deserves one test there.

added

File: benchto-it/src/test/resources/benchmarks/test_throughput_test.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+datasource: test_datasource
+query-names: ${query}.sql
```
Comment (przemekak) [inline benchto-it/src/test/resources/benchmarks/test_throughput_test.yaml:2]: nit: make it a little more as pseudo throughput test? 
I mean use short query-names list, with concurrency and runs > 3?

**Review by przemekak — APPROVED** [review]

File: benchto-it/src/test/resources/benchmarks/test_throughput_test.yaml (hunk 3 above)
Comment (radek-starburst) [reply benchto-it/src/test/resources/benchmarks/test_throughput_test.yaml:2]: > I mean use short query-names list
no difference on my eye
> concurrency and runs > 4
done

File: benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java [hunk 4]
```diff
@@ -249,7 +249,7 @@ private List<QueryExecutionResult> executeConcurrentQueries(Benchmark benchmark,
                         firstQuery = false;
                     }
                     try {
-                        queryExecutionResults.add(executeSingleQuery(queryExecution, benchmark, connection, false, executionTimeLimit));
+                        queryExecutionResults.add(executeSingleQuery(queryExecution, benchmark, connection, warmup, executionTimeLimit));
```
Comment (sopel39) [inline benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java:252]: Why this change is needed since you added `&& !benchmark.isThroughputTest()`?

Wouldn't you get same result by leaving `false` here?

File: benchto-it/src/test/java/io/trino/benchto/integrationtest/BenchtoTrinoIntegrationTest.java [hunk 5]
```diff
@@ -228,12 +246,7 @@ private void verifyBenchmark(String benchmark, String status, int expectedRuns)
         verifyBenchmark(benchmark, benchmark, status, expectedRuns);
     }
 
-    private void verifyBenchmark(String benchmark, String sequenceId, String status)
-    {
-        verifyBenchmark(benchmark, sequenceId, status, 2);
-    }
-
-    private void verifyBenchmark(String benchmark, String sequenceId, String status, int expectedRuns)
+    private void verifyBenchmark(String benchmark, String sequenceId, String status, int expectedRuns, Predicate<DocumentContext> predicate)
```
Comment (sopel39) [inline benchto-it/src/test/java/io/trino/benchto/integrationtest/BenchtoTrinoIntegrationTest.java:249]: move this method as last since it's used by every other method

File: benchto-it/src/test/java/io/trino/benchto/integrationtest/BenchtoTrinoIntegrationTest.java [hunk 6]
```diff
@@ -228,12 +246,7 @@ private void verifyBenchmark(String benchmark, String status, int expectedRuns)
         verifyBenchmark(benchmark, benchmark, status, expectedRuns);
     }
 
-    private void verifyBenchmark(String benchmark, String sequenceId, String status)
```
Comment (sopel39) [inline benchto-it/src/test/java/io/trino/benchto/integrationtest/BenchtoTrinoIntegrationTest.java:231]: undo

File: benchto-it/src/test/java/io/trino/benchto/integrationtest/BenchtoTrinoIntegrationTest.java [hunk 7]
```diff
@@ -145,6 +147,22 @@ public void testVerifyInsertResultsFailureWithPreWarm()
...
+    {
+        setBenchmark("test_throughput_test");
+        executionDriver.execute();
+        verifyBenchmark(
+                "test_throughput_test_query=test_results",
+                "test_throughput_test",
+                "ENDED",
+                4,
+                document -> {
+                    List<Map<String, Object>> successFullQueries = document.read("$['executions'][*]['measurements'][*][?(@.name==\"queries_successful\")]");
```
Comment (sopel39) [inline benchto-it/src/test/java/io/trino/benchto/integrationtest/BenchtoTrinoIntegrationTest.java:161]: `successFullQueries` -> `successfullQueries` ?

add comment what it tests

File: benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java (hunk 4 above)
Comment (radek-starburst) [reply benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java:252]: If you set `warmup` to `false` you have at least two bugs in two different code paths:
1. Code is trying to report that **benchmark execution** was started despited of the fact **benchmark** (it is a higher level term) is not started. It is a one problem.
2. The second one is when we try to report `executionStarted` twice. 
 
If we set the `warmup` to the `true` it also resolved the issue. But, it is not a good idea to lie because, as you see, it is very errorneous to make things implicitly instead of explicitly like here: 
"Do not report that stuff for `throughputTest`" instead of trying to use another parameter in specific way.

File: benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java [hunk 8]
```diff
@@ -284,7 +284,7 @@ private QueryExecutionResult executeSingleQuery(
         QueryExecutionResult result;
         macroService.runBenchmarkMacros(benchmark.getBeforeExecutionMacros(), benchmark, connection);
 
-        if (!warmup) {
```
Comment (sopel39) [inline benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java:287]: `warmup` -> `skipReport`

File: benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java (hunk 4 above)
Comment (sopel39) [inline benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java:252]: Just set it to `true` + comment

File: benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java (hunk 2 above)
Comment (sopel39) [inline benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java:287]: this is not needed

**Review by sopel39 — COMMENTED** [review]
lgtm % comments

**Review by sopel39 — APPROVED** [review]
